src/transformers/pipelines/document_question_answering.py

```diff
@@ -399,41 +433,38 @@ def postprocess_donut(self, model_outputs, **kwargs):
         answer = re.search(r"<s_answer>(.*)</s_answer>", sequence)
         if answer is not None:
             ret["answer"] = answer.group(1).strip()
-        return [ret]
+        return ret
 
     def postprocess_extractive_qa(
         self, model_outputs, top_k=1, handle_impossible_answer=False, max_answer_len=15, **kwargs
     ):
         min_null_score = 1000000  # large and positive
         answers = []
-        words = model_outputs["words"]
-
-        # TODO: Currently, we expect the length of model_outputs to be 1, because we do not stride
-        # in the preprocessor code. When we implement that, we'll either need to handle tensors of size
-        # > 1 or use the ChunkPipeline and handle multiple outputs (each of size = 1).
-        starts, ends, scores, min_null_score = select_starts_ends(
-            model_outputs["start_logits"],
-            model_outputs["end_logits"],
-            model_outputs["p_mask"],
-            model_outputs["attention_mask"].numpy() if model_outputs.get("attention_mask", None) is not None else None,
-            min_null_score,
-            top_k,
-            handle_impossible_answer,
-            max_answer_len,
-        )
-
-        word_ids = model_outputs["word_ids"][0]
-        for start, eend, score in zip(starts, ends, scores):
-            word_start, word_end = word_ids[start], word_ids[eend]
-            if word_start is not None and word_end is not None:
-                answers.append(
-                    {
-                        "score": float(score),  # XXX Write a test that verifies the result is JSON-serializable
-                        "answer": " ".join(words[word_start : word_end + 1]),
-                        "start": word_start,
-                        "end": word_end,
-                    }
-                )
+        for output in model_outputs:
+            words = output["words"]
+
+            starts, ends, scores, min_null_score = select_starts_ends(
+                output["start_logits"],
+                output["end_logits"],
+                output["p_mask"],
+                output["attention_mask"].numpy() if output.get("attention_mask", None) is not None else None,
+                min_null_score,
+                top_k,
+                handle_impossible_answer,
+                max_answer_len,
+            )
```

Comment:
Nit, using keyword arguments here would make the code easier to maintain.

Reply:
I can change that! Good idea.